doc/query.md

```diff
@@ -0,0 +1,545 @@
+successful.
+
+However, if there was one index with `pet` in and another with `name` in, like this:
+
+```java
+String indexOne = im.ensureIndexed(Arrays.<Object>asList("name", "age"), 
+                                   "index_one");
+String indexTwo = im.ensureIndexed(Arrays.<Object>asList("age", "pet"),
+                                   "index_two");
+```
+
+The document _would_ be indexed in both of these indexes: each index only contains one of
+the array fields.
+
+Also see "Unsupported features", below.
+
+
+### Errors
+
+Error reporting is terrible right now. The only indication something went wrong is a `null` return value from the `find` methods or `ensureIndexed(List<Object> fieldNames, String indexName)`. We're working on
+adding logging.
+
+## Supported Cloudant Query features
+
+Right now the list of supported features is:
+
+- Create compound indexes using dotted notation that index JSON fields.
+- Delete index by name.
+- Execute nested queries.
+- Limiting returned results.
+- Skipping results.
+- Queries can include unindexed fields.
+      
+Selectors -> combination
+
+- `$and`
+- `$or`
+
+Selectors -> Conditions -> Equalities
+
+- `$lt`
+- `$lte`
+- `$eq`
+- `$gte`
+- `$gt`
+- `$ne`
+
+Selectors -> combination
+
+- `$not`
+
+Selectors -> Condition -> Objects
+
+- `$exists`
+
+Implicit operators
+
+- Implicit `$and`.
+- Implicit `$eq`.
+
+Arrays
+
+- Indexing individual values in an array.
+- Querying for individual values in an array.
+
+## Unsupported Cloudant Query features
+
+As this is an early version of Query on this platform, some features are
+not supported yet. We're actively working to support features -- check
+the commit log :)
+
+### Query
+
+Overall restrictions:
+
+- Cannot use covering indexes with projection (`fields`) to avoid loading 
+  documents from the datastore.
+
+#### Query syntax
+
+- Using non-dotted notation to query sub-documents.
+    - That is, `{"pet": { "species": {"$eq": "cat"} } }` is unsupported,
+      you must use `{"pet.species": {"$eq": "cat"}}`.
+- Cannot use multiple conditions in a single clause, `{ field: { $gt: 7, $lt: 14 } }`.
+
+Selectors -> combination
+
+- `$nor` #10
+- `$all` (unplanned)
+- `$elemMatch` (unplanned, waiting on arrays support for query)
+
+Selectors -> Condition -> Objects
+
+- `$type` (unplanned)
+
+Selectors -> Condition -> Array
+
+- `$in` (waiting on arrays support)
+- `$nin` (waiting on arrays support)
+- `$size` (waiting on arrays support)
+
+Selectors -> Condition -> Misc
+
+- `$mod` (unplanned, waiting on filtering)
+- `$regex` (unplanned, waiting on filtering)
+
+
+Arrays
+
+- Dotted notation to index or query sub-documents in arrays.
+- Querying for exact array match, `{ field: [ 1, 3, 7 ] }`.
+- Querying to match a specific array element using dotted notation, `{ field.0: 1 }`.
+- Querying using `$all`.
+- Querying using `$elemMatch`.
+
+
+## Performance
+
+### Indexing
+
+N/A
```

Comment:
I was using N/A as Not Available.  Updated to "Not carried out yet" to remove any ambiguity https://github.com/cloudant/sync-android/commit/d5b4f7825035187c3ef55b5962f38b66806f68d3